src/Mint.js

```diff
@@ -0,0 +1,354 @@
+import TheFacesFund from './abi/TheFacesFund.json';
+
+const defaultErrorTitle = 'Whoops!';
+const defaultErrorMessage = 'Looks like something went wrong! If this issue persists please reach out to the team on Discord and we will get back to you ASAP.';
+
+const Mint = () => {
+	// Wallet state
+	const [signedIn, setSignedIn] = useState(false);
+	const [walletAddress, setWalletAddress] = useState(null);
+
+	// Minting state
+	const [facesFundContract, setFacesFundContract] = useState(null);
+	const [numTokensToPurchase, setNumTokensToPurchase] = useState(1);
+	const [validNumTokens, setValidNumTokens] = useState(true);
+	const [showNote, setShowNote] = useState(false);
+	const [errorTitle, setErrorTitle] = useState(defaultErrorTitle);
+	const [errorMessage, setErrorMessage] = useState(defaultErrorMessage);
+
+	// Current contract data
+	const [totalSupply, setTotalSupply] = useState(0);
+	const [saleStarted, setSaleStarted] = useState(false);
+	const [tokenPrice, setTokenPrice] = useState(0);
+
+	const signIn = async () => {
+		// Check if metamask is installed.
+		// TODO: Check once metamask is installed or instruct to refresh page
+		if (window.ethereum) {
+			if (typeof window.web3 !== 'undefined') {
+				// Use existing gateway
+				window.web3 = new Web3(window.ethereum);
+	
+				try {
+					const accounts = await window.ethereum.request({ method: 'eth_requestAccounts' });
+					const network = await window.web3.eth.net.getNetworkType();
+
+					// Set to desired network i.e. "rinkeby" for testing
+					if (network !== "rinkeby") {
+						console.log(network);
+						alert("You are on " + network + " network. Change network to mainnet or you won't be able to do anything here");
+					}
+					else {
+						let wallet = accounts[0];
+						setWalletAddress(wallet);
+						setSignedIn(true);
+						getContractData(wallet);
+					}
+				}
+				catch (e) {
+					// TODO: Handle error.
+					if (e.code === 4001) {
+						// User rejected request
+						setErrorMessage('It looks like your wallet connection request failed! If you would like to mint a token please try again.');
+						console.log('User rejected request:', e);
+					}
+					else if (e.code === -32002) {
+						setErrorMessage('It looks like there was an issue connecting your wallet (you may have accidentally exited out of your connection window). Please refresh the page and try again. If this issue persists please reach out to our team on Discord and we will get back to you ASAP!');
+					}
+					else {
+						console.error(e);
+					}
+					setShowNote(true);
+				}
+			} else {
+				setErrorMessage("No Ethereum interface injected into browser. Make sure you have the Metamask extension installed!");
+				setShowNote(true);
+			}
+		}
+		else {
+			setErrorMessage('It looks like you may not have Metamask installed. Please see the instructions below to connect your wallet.');
+			setShowNote(true);
+		}
+	}
+
+	// Do we need a sign out function?
```

Comment:
I think a user can just disconnect their MetaMask from our website if they want to sign out

Reply:
Yeah after some research this isn't really a thing. There are ways to do it but for now i'll ignore it